manifests/init.pp

```diff
@@ -39,9 +39,16 @@
     purge_values => $purge_values
   }
 
-  service{ 'wuauserv':
-    ensure => running,
-    enable => true,
+  case $::kernelmajversion {
```

Comment:
Is the kernel check is always the right thing to do?  What about future kernels, etc?

FYI - on my 2008R2 (kernel 6.1) that's supposed to support trigger starts, I get this:

```
C:\Users\Administrator>puppet resource service wuauserv
service { 'wuauserv':
  ensure => 'running',
  enable => 'true',
}

C:\Users\Administrator>sc queryex wuauserv

SERVICE_NAME: wuauserv
        TYPE               : 20  WIN32_SHARE_PROCESS
        STATE              : 4  RUNNING
                                (STOPPABLE, NOT_PAUSABLE, ACCEPTS_PRESHUTDOWN)
        WIN32_EXIT_CODE    : 0  (0x0)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x0
        PID                : 876
        FLAGS              :

C:\Users\Administrator>sc qtriggerinfo wuauserv
[SC] QueryServiceConfig2 SUCCESS

    The service wuauserv has not registered for any start or stop triggers.
```

Note that I haven't used the module to configure WSUS settings though.

On Windows 10, in contrast:

```
$ sc qtriggerinfo wuauserv
[SC] QueryServiceConfig2 SUCCESS

SERVICE_NAME: wuauserv

        START SERVICE
          GROUP POLICY                 : 659fcae6-5bdb-4da9-b1ff-ca2a178d46e0 [MACHINE POLICY PRESENT]
        START SERVICE
          GROUP POLICY                 : 54fb46c8-f089-464c-b1fd-59d1b62c3b50 [USER POLICY PRESENT]
```

Checking these values (or the equivalent registry settings) is probably a better guard than kernelversion.